pkg/operator/configobservation/images/observe_images_test.go

```diff
@@ -45,3 +46,46 @@ func TestObserveRegistryConfig(t *testing.T) {
 		t.Errorf("expected internal registry hostname: %s, got %s", expectedInternalRegistryHostname, internalRegistryHostname)
 	}
 }
+
+func TestObserveRegistryExternalRoutes(t *testing.T) {
+	expectedExternalRegistryHostname := []string{
+		"default-route-0.openshift-image-registry.io",
+		"default-route-1.openshift-image-registry.io",
+	}
+
+	indexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+	imageConfig := &configv1.Image{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "cluster",
+		},
+		Spec: configv1.ImageSpec{
+			ExternalRegistryHostnames: []string{
+				"default-route-2.openshift-image-registry.io",
+			},
+		},
+		Status: configv1.ImageStatus{
+			ExternalRegistryHostnames: []string{
+				"default-route-0.openshift-image-registry.io",
+				"default-route-1.openshift-image-registry.io",
+			},
+		},
+	}
```

Comment:
Please add the following test cases:

1. No external hostname (ensure we don't populate the Registry URLs)
2. External hostnames in `Spec`, but not `Status` (ensure we don't populate the registry URLs)
3. External hostnames in `Status`, but not `Spec` (ensure we keep the registry URLs)

Table-driven testing is highly effective for this purpose, and Golang testing makes it very easy to write:

https://dave.cheney.net/2019/05/07/prefer-table-driven-tests

Reply:
done, PTAL.